src/HtmlUtils.tsx

```diff
@@ -302,6 +306,16 @@ function analyseEvent(content: IContent, highlights: Optional<string[]>, opts: E
         sanitizeParams = composerSanitizeHtmlParams;
     }
 
+    if (opts.mediaIsVisible === false && sanitizeParams.transformTags?.["img"]) {
+        // Prevent mutating the source of sanitizeParams.
+        sanitizeParams.transformTags = {
+            ...sanitizeParams.transformTags,
+            img: (tagName) => {
+                return { tagName, attribs: {} };
+            },
+        };
```

Comment:
this still looks like it mutates the original `sanitizeHtmlParams` or `composerSanitizeHtmlParams`